packages/celotool/src/lib/generate_utils.ts

```diff
@@ -83,6 +85,24 @@ export const generatePrivateKey = (mnemonic: string, accountType: AccountType, i
   return generatePrivateKeyWithDerivations(mnemonic, [accountType, index])
 }
 
+export const generateOraclePrivateKey = (
+  mnemonic: string,
+  currencyPair: CurrencyPair,
+  index: number
+) => {
+  let path: number[]
+  if (currencyPair === 'CELOUSD') {
+    // For backwards compatibility we don't add currencyPair to 
+    // the derivation path for CELOUSD
+    path = [AccountType.PRICE_ORACLE, index]
+  } else {
+    const currencyDerivation = new BigNumber(Web3.utils.keccak256(currencyPair), 16).mod(2 ** 31).toNumber()
+    path = [AccountType.PRICE_ORACLE, currencyDerivation, index]
```

Comment:
nit: shall we add a comment around some of the magic constants? 

Also do you see any major downside of just using 0,1,2,...n as the 2nd item in the path depending on the currencyPair, similar to AccountType?

Reply:
When you say magic constant do you mean the 2nd path segment? - then yeah I can do it but I don't see what extra information I can add in the comment that's not just a verbatim description of what's happening in code.
A la: "Convert the currency pair to an int % 2 ** 31 because that's the max value for a derivation path segment"

Also it might be a matter of personal preference but going with this makes more sense as currency pair is conceptually more unbounded than `AccountType`. So for me it feels better to have a clear mechanism to easily derive pks for arbitrary oracles, rather than always having to edit the enum and add a new value to it when the need arises. But that being said, I'm not married to it and curious to hear what the upsides of using a static enum are.

Reply:
I guess you can see it as creating an unbounded deterministic procedural mapping vs hardcoding a bounded mapping.

Reply:
Thanks for sharing your thoughts, I don't have any strong feelings about this either and I'm happy to go with your approach, I was mostly curious to hear your reasoning. 

Personally I would have probably gone with the enum as it seems more straightforward to me code-wise (with the exception that I don't think we have that enum somewhere else, but if that was already the case then I think that approach would make even more sense). I didn't have that unbounded vs bounded consideration as I don't see this growing a lot relatively soon.